packages/lambdas/src/shared/converter.ts

```diff
@@ -0,0 +1,211 @@
+capture.init();
+
+// const converterKeysTableName = getEnvOrFail("FHIR_T0_CDA_CONVERTER_KEYS_TABLE_NAME");
+enum converterKeysStatus {
+  active = "active",
+  rateLimit = "rate-limit",
+  revoked = "revoked",
+}
+const MAX_CONVERTER_ATTEMPTS = 5;
+const CONVERTER_INITIAL_TIME_BETTWEEN_ATTEMPTS_MILLIS = 500;
+
+const createFhirConverter = (axiosTimeoutSeconds: number) =>
+  axios.create({
+    // Only response timeout, no option for connection timeout: https://github.com/axios/axios/issues/4835
+    timeout: axiosTimeoutSeconds * 1_000, // should be less than the lambda timeout
+    transitional: {
+      // enables ETIMEDOUT instead of ECONNABORTED for timeouts - https://betterstack.com/community/guides/scaling-nodejs/nodejs-errors/
+      clarifyTimeoutError: true,
+    },
+  });
+
+export async function postToConverter({
+  url,
+  payload,
+  axiosTimeoutSeconds,
+  converterKeysTableName,
+  log,
+  contentType,
+}: {
+  url: string;
+  payload: unknown;
+  axiosTimeoutSeconds: number;
+  converterKeysTableName: string;
+  log: Log;
+  contentType?: string;
+}) {
+  const sidechainUrl = url;
+  let attempt = 0;
+  let timeBetweenAttemptsMillis = CONVERTER_INITIAL_TIME_BETTWEEN_ATTEMPTS_MILLIS;
+  let apiKey: string;
+  while (attempt++ < MAX_CONVERTER_ATTEMPTS) {
+    apiKey = await getAndUpdateSidechainConverterKeys(converterKeysTableName);
+    log(`(${attempt}) Calling sidechain converter on url ${sidechainUrl}`);
+    try {
+      const fhirConverter = createFhirConverter(axiosTimeoutSeconds);
+
+      const res = await fhirConverter.post(sidechainUrl, payload, {
+        headers: {
+          ...(contentType ? { "Content-Type": contentType } : {}),
+          Accept: "application/json",
+          "x-api-key": apiKey,
+        },
+      });
+      return res;
+      //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      if ([401, 429].includes(error.response?.status)) {
+        const msg = "Sidechain quota/auth error, trying again";
+        const extra = {
+          url: sidechainUrl,
+          apiKey,
+          statusCode: error.response?.status,
+          attempt,
+          error,
+        };
+        log(msg, extra);
+        capture.message(msg, { extra, level: "info" });
+        if (error.response.status === 429) {
+          await markSidechainConverterKeyAsRateLimited(apiKey, converterKeysTableName);
+        } else {
+          await markSidechainConverterKeyAsRevoked(apiKey, converterKeysTableName);
+        }
+        await sleep(timeBetweenAttemptsMillis);
+        timeBetweenAttemptsMillis *= 2;
+      } else {
+        throw error;
+      }
+    }
+  }
+  throw new Error(`Too many errors from sidechain converter`);
+}
+
+async function getAndUpdateSidechainConverterKeys(converterKeysTableName: string): Promise<string> {
+  if (!converterKeysTableName) {
+    throw new Error(`Programming error - FHIR_T0_CDA_CONVERTER_KEYS_TABLE_NAME is not set`);
```

Comment:
shouldn't this env var be named `SIDECHAIN_FHIR_CONVERTER_KEYS_TABLE_NAME` so this code can be reused in both lambdas?